src/event/event_data/rotation.rs

```diff
@@ -17,4 +19,17 @@ pub struct RotationEvent {
     pub witness_config: WitnessConfig,
 }
 
-impl EventSemantics for RotationEvent {}
+impl EventSemantics for RotationEvent {
+    fn apply_to(&self, state: IdentifierState) -> Result<IdentifierState, Error> {
+        Ok(IdentifierState {
+            last: self.previous_event_hash.clone(),
```

Comment:
I think I made a mistake with the type of this field, `last` should probably be the fully serialised previous event instead of the hash, as the type of the previous event hash included in the next event may not be known yet. then `self.previous_event_hash.derivation.derive(last) == self.previous_event_hash` can be checked. the relevant fix may be a separate issue to #18